src/ray/object_manager/object_manager.cc

```diff
@@ -191,13 +196,27 @@ ray::Status ObjectManager::PullSendRequest(const ObjectID &object_id,
 }
 
 ray::Status ObjectManager::Push(const ObjectID &object_id, const ClientID &client_id) {
-  // TODO(hme): Cache this data in ObjectDirectory.
-  // Okay for now since the GCS client caches this data.
+  if (local_objects_.count(object_id) == 0) {
+    // TODO(hme): Do not retry indefinitely...
+    main_service_->post([this, object_id, client_id]() { Push(object_id, client_id); });
+    return ray::Status::OK();
```

Comment:
Was this causing errors in this PR? Also, does it make sense to return `OK` here?